### ISSUE TITLE
ESP-IDF 5.2.1 issue with installed FMT in VSC (Window 11), it generate missing CPP/G++ header error, unable to complies

### ISSUE THREAD

**Issue by riscy00** [google/oss-fuzz#11821]
Working on ESP32-H2 under ESP-IDF 5.2.1 which Expressif/FMT is installed into a project under VSC (Window 11) via add component feature of the ESP-IDF (extension). 
It goes to managed_components, according to the compiler logs, it does add g and g++ but it generate error of missing g++ library,
Not sure if this is bug or missing something in cmake.
I have move FMT in managed component into components folder but it does not solve compiler problem (same error as in managed components)  
In the gpio_example_main.c : #include "fmt/core.h" generate error below. When I comment this out it complies without error. 

In file included from E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/main/gpio_example_main.c:18:
E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/components/espressif__fmt/fmt/include/fmt/core.h:11:10: fatal error: cstddef: No such file or directory
   11 | #include <cstddef>  // std::byte
      |          ^~~~~~~~~
compilation terminated.

-- Found Git: C:/Users/riscy/.espressif/tools/idf-git/2.39.2/cmd/git.exe (found version "2.39.2.windows.1") 
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-gcc.exe
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- git rev-parse returned 'fatal: not a git repository (or any of the parent directories): .git'
-- Could not use 'git describe' to determine PROJECT_VER.
-- Building ESP-IDF components for target esp32h2
Processing 1 dependencies:
[1/1] idf (5.2.1)
-- Project sdkconfig file E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/sdkconfig
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Found Python3: C:/Users/riscy/.espressif/python_env/idf5.2_py3.11_env/Scripts/python.exe (found version "3.11.2") found components: Interpreter 
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS
-- Performing Test C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS - Success
-- App "TestH2_GPIO_1AX" version: 1
-- Adding linker script E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/build/esp-idf/esp_system/ld/memory.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system/ld/esp32h2/sections.ld.in
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.api.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.libgcc.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.newlib.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.version.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.wdt.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.heap.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/soc/esp32h2/ld/esp32h2.peripherals.ld
-- Version: 10.1.0
-- Build type:
-- Components: app_trace app_update bootloader bootloader_support bt cmock console cxx driver efuse esp-tls esp_adc esp_app_format esp_bootloader_format esp_coex esp_common esp_eth esp_event esp_gdbstub esp_hid esp_http_client esp_http_server esp_https_ota esp_https_server esp_hw_support esp_lcd esp_local_ctrl esp_mm esp_netif esp_netif_stack esp_partition esp_phy esp_pm esp_psram esp_ringbuf esp_rom esp_system esp_timer esp_wifi espcoredump espressif__fmt esptool_py fatfs freertos hal heap http_parser idf_test ieee802154 json log lwip main mbedtls mqtt newlib nvs_flash nvs_sec_provider openthread partition_table protobuf-c protocomm pthread riscv sdmmc soc spi_flash spiffs tcp_transport ulp unity usb vfs wear_levelling wifi_provisioning wpa_supplicant
-- Component paths: C:/Users/riscy/esp/v5.2.1/esp-idf/components/app_trace C:/Users/riscy/esp/v5.2.1/esp-idf/components/app_update C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader_support C:/Users/riscy/esp/v5.2.1/esp-idf/components/bt C:/Users/riscy/esp/v5.2.1/esp-idf/components/cmock C:/Users/riscy/esp/v5.2.1/esp-idf/components/console C:/Users/riscy/esp/v5.2.1/esp-idf/components/cxx C:/Users/riscy/esp/v5.2.1/esp-idf/components/driver C:/Users/riscy/esp/v5.2.1/esp-idf/components/efuse C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp-tls C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_adc C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_app_format C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_bootloader_format C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_coex C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_common C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_eth C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_event C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_gdbstub C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hid C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_http_client C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_http_server C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_https_ota C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_https_server C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_lcd C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_local_ctrl C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_mm C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_netif C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_netif_stack C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_partition C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_phy C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_pm C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_psram C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_ringbuf C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_timer C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_wifi C:/Users/riscy/esp/v5.2.1/esp-idf/components/espcoredump E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/components/espressif__fmt C:/Users/riscy/esp/v5.2.1/esp-idf/components/esptool_py C:/Users/riscy/esp/v5.2.1/esp-idf/components/fatfs C:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos C:/Users/riscy/esp/v5.2.1/esp-idf/components/hal C:/Users/riscy/esp/v5.2.1/esp-idf/components/heap C:/Users/riscy/esp/v5.2.1/esp-idf/components/http_parser C:/Users/riscy/esp/v5.2.1/esp-idf/components/idf_test C:/Users/riscy/esp/v5.2.1/esp-idf/components/ieee802154 C:/Users/riscy/esp/v5.2.1/esp-idf/components/json C:/Users/riscy/esp/v5.2.1/esp-idf/components/log C:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/main C:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls C:/Users/riscy/esp/v5.2.1/esp-idf/components/mqtt C:/Users/riscy/esp/v5.2.1/esp-idf/components/newlib C:/Users/riscy/esp/v5.2.1/esp-idf/components/nvs_flash C:/Users/riscy/esp/v5.2.1/esp-idf/components/nvs_sec_provider C:/Users/riscy/esp/v5.2.1/esp-idf/components/openthread C:/Users/riscy/esp/v5.2.1/esp-idf/components/partition_table C:/Users/riscy/esp/v5.2.1/esp-idf/components/protobuf-c C:/Users/riscy/esp/v5.2.1/esp-idf/components/protocomm C:/Users/riscy/esp/v5.2.1/esp-idf/components/pthread C:/Users/riscy/esp/v5.2.1/esp-idf/components/riscv C:/Users/riscy/esp/v5.2.1/esp-idf/components/sdmmc C:/Users/riscy/esp/v5.2.1/esp-idf/components/soc C:/Users/riscy/esp/v5.2.1/esp-idf/components/spi_flash C:/Users/riscy/esp/v5.2.1/esp-idf/components/spiffs C:/Users/riscy/esp/v5.2.1/esp-idf/components/tcp_transport C:/Users/riscy/esp/v5.2.1/esp-idf/components/ulp C:/Users/riscy/esp/v5.2.1/esp-idf/components/unity C:/Users/riscy/esp/v5.2.1/esp-idf/components/usb C:/Users/riscy/esp/v5.2.1/esp-idf/components/vfs C:/Users/riscy/esp/v5.2.1/esp-idf/components/wear_levelling C:/Users/riscy/esp/v5.2.1/esp-idf/components/wifi_provisioning C:/Users/riscy/esp/v5.2.1/esp-idf/components/wpa_supplicant
-- Configuring done
-- Generating done
-- Build files have been written to: E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/build

 *  Executing task: ninja  

[20/933] Generating ../../partition_table/partition-table.bin
Partition table binary generated. Contents:
*******************************************************************************
# ESP-IDF Partition Table
# Name, Type, SubType, Offset, Size, Flags
nvs,data,nvs,0x9000,24K,
phy_init,data,phy,0xf000,4K,
factory,app,factory,0x10000,1M,
*******************************************************************************
[680/933] Performing configure step for 'bootloader'
-- Found Git: C:/Users/riscy/.espressif/tools/idf-git/2.39.2/cmd/git.exe (found version "2.39.2.windows.1")
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- The ASM compiler identification is GNU
-- Found assembler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-gcc.exe
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Users/riscy/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20230928/riscv32-esp-elf/bin/riscv32-esp-elf-g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Building ESP-IDF components for target esp32h2
-- Project sdkconfig file E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/sdkconfig
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of time_t
-- Check size of time_t - done
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/soc/esp32h2/ld/esp32h2.peripherals.ld
-- Bootloader project name: "bootloader" version: 1
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.api.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.libgcc.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.newlib.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2/ld/esp32h2.rom.wdt.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader/subproject/main/ld/esp32h2/bootloader.ld
-- Adding linker script C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader/subproject/main/ld/esp32h2/bootloader.rom.ld
-- Components: bootloader bootloader_support efuse esp_app_format esp_bootloader_format esp_common esp_hw_support esp_rom esp_system esptool_py freertos hal log main micro-ecc newlib partition_table riscv soc spi_flash
-- Component paths: C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader_support C:/Users/riscy/esp/v5.2.1/esp-idf/components/efuse C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_app_format C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_bootloader_format C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_common C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom C:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system C:/Users/riscy/esp/v5.2.1/esp-idf/components/esptool_py C:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos C:/Users/riscy/esp/v5.2.1/esp-idf/components/hal C:/Users/riscy/esp/v5.2.1/esp-idf/components/log C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader/subproject/main C:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader/subproject/components/micro-ecc C:/Users/riscy/esp/v5.2.1/esp-idf/components/newlib C:/Users/riscy/esp/v5.2.1/esp-idf/components/partition_table C:/Users/riscy/esp/v5.2.1/esp-idf/components/riscv C:/Users/riscy/esp/v5.2.1/esp-idf/components/soc C:/Users/riscy/esp/v5.2.1/esp-idf/components/spi_flash
-- Configuring done
-- Generating done
-- Build files have been written to: E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/build/bootloader
[900/933] Building C object esp-idf/main/CMakeFiles/__idf_main.dir/gpio_example_main.c.obj
FAILED: esp-idf/main/CMakeFiles/__idf_main.dir/gpio_example_main.c.obj
C:\Users\riscy\.espressif\tools\riscv32-esp-elf\esp-13.2.0_20230928\riscv32-esp-elf\bin\riscv32-esp-elf-gcc.exe -DESP_PLATFORM -DIDF_VER=\"v5.2.1\" -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -DSOC_MMU_PAGE_SIZE=CONFIG_MMU_PAGE_SIZE -DSOC_XTAL_FREQ_MHZ=CONFIG_XTAL_FREQ -DUNITY_INCLUDE_CONFIG_H -D_GLIBCXX_HAVE_POSIX_SEMAPHORE -D_GLIBCXX_USE_POSIX_SEMAPHORE -D_GNU_SOURCE -D_POSIX_READER_WRITER_LOCKS -IE:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/build/config -IE:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/main -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/newlib/platform_include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/config/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/config/include/freertos -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/config/riscv/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/FreeRTOS-Kernel/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/FreeRTOS-Kernel/portable/riscv/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/FreeRTOS-Kernel/portable/riscv/include/freertos -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/freertos/esp_additions/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support/include/soc -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support/include/soc/esp32h2 -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support/port/esp32h2/. -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hw_support/port/esp32h2/private_include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/heap/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/log/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/soc/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/soc/esp32h2 -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/soc/esp32h2/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/hal/platform_port/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/hal/esp32h2/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/hal/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/include/esp32h2 -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_rom/esp32h2 -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_common/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system/port/soc -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system/port/include/riscv -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_system/port/include/private -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/riscv/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/include/apps -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/include/apps/sntp -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/lwip/src/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/port/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/port/freertos/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/port/esp32xx/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/port/esp32xx/include/arch -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/lwip/port/esp32xx/include/sys -IE:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/components/espressif__fmt/fmt/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_ringbuf/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/efuse/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/efuse/esp32h2/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_mm/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/deprecated -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/analog_comparator/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/dac/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/gpio/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/gptimer/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/i2c/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/i2s/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/ledc/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/mcpwm/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/parlio/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/pcnt/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/rmt/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/sdio_slave/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/sdmmc/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/sigma_delta/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/spi/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/temperature_sensor/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/touch_sensor/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/twai/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/uart/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/driver/usb_serial_jtag/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_pm/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/port/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/mbedtls/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/mbedtls/library -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/esp_crt_bundle/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/mbedtls/3rdparty/everest/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/mbedtls/3rdparty/p256-m -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mbedtls/mbedtls/3rdparty/p256-m/p256-m -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_bootloader_format/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_app_format/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader_support/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/bootloader_support/bootloader_flash/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_partition/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/app_update/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/spi_flash/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/pthread/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_timer/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/app_trace/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_event/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/nvs_flash/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_phy/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_phy/esp32h2/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/vfs/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_netif/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/wpa_supplicant/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/wpa_supplicant/port/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/wpa_supplicant/esp_supplicant/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_coex/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_wifi/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_wifi/wifi_apps/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/unity/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/unity/unity/src -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/cmock/CMock/src -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/console -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/http_parser -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp-tls -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp-tls/esp-tls-crypto -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_adc/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_adc/interface -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_adc/esp32h2/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_adc/deprecated/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_eth/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_gdbstub/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_hid/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/tcp_transport/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_http_client/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_http_server/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_https_ota/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_psram/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_lcd/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_lcd/interface -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/protobuf-c/protobuf-c -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/protocomm/include/common -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/protocomm/include/security -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/protocomm/include/transports -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/protocomm/include/crypto/srp6a -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/esp_local_ctrl/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/espcoredump/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/espcoredump/include/port/riscv -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/wear_levelling/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/sdmmc/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/fatfs/diskio -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/fatfs/src -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/fatfs/vfs -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/idf_test/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/idf_test/include/esp32h2 -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/ieee802154/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/json/cJSON -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/mqtt/esp-mqtt/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/nvs_sec_provider/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/spiffs/include -IC:/Users/riscy/esp/v5.2.1/esp-idf/components/wifi_provisioning/include -march=rv32imac_zicsr_zifencei  -fdiagnostics-color=always -ffunction-sections -fdata-sections -Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=unused-but-set-variable -Wno-error=deprecated-declarations -Wextra -Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -nostartfiles -Og -fno-shrink-wrap -fmacro-prefix-map=E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX=. -fmacro-prefix-map=C:/Users/riscy/esp/v5.2.1/esp-idf=/IDF -fstrict-volatile-bitfields -fno-jump-tables -fno-tree-switch-conversion -std=gnu17 -Wno-old-style-declaration -MD -MT esp-idf/main/CMakeFiles/__idf_main.dir/gpio_example_main.c.obj -MF esp-idf\main\CMakeFiles\__idf_main.dir\gpio_example_main.c.obj.d -o esp-idf/main/CMakeFiles/__idf_main.dir/gpio_example_main.c.obj -c E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/main/gpio_example_main.c
In file included from E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/main/gpio_example_main.c:18:
E:/016_Work_ESP32/010_Example/BlinkH2_ESPIDF/WorkspaceH2/TestH2_GPIO_1A/TestH2_GPIO_1AX/components/espressif__fmt/fmt/include/fmt/core.h:11:10: fatal error: cstddef: No such file or directory
   11 | #include <cstddef>  // std::byte
      |          ^~~~~~~~~
compilation terminated.
[904/933] Performing build step for 'bootloader'
[1/106] Generating project_elf_src_esp32h2.c

**Comment by riscy00** [comment]
I moved the fmt code from main.c to separate file fmttest.cpp with #include "fmt/core.h" and wrote a function fmttestrun() that use fmt to print simple message. I found the above error goes away and passed complier but linker failed as it could not find fmttestrun() so it back to square one. I still learning mixed c and cpp between them. VSC is very different from eclipse-based IDE (STM32Cube IDE, Code composer, etc).
